src/NuGet.Clients/NuGet.PackageManagement.UI/UserInterfaceService/VisualStudioProgressDialogSession.cs

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using Microsoft.VisualStudio.Shell;
+
+namespace NuGet.PackageManagement.UI
+{
+    internal class VisualStudioProgressDialogSession : IModalProgressDialogSession
+    {
+        private readonly Action _dispose;
+
```

Comment:
add `bool _isDisposed`, to make sure the dispose action is not run twice

Reply:
is that even possible?